File: tests/smoke/send.pl
#!/usr/bin/env perl
require 'qpid_proton.pm';
my $messenger = qpid::proton::Messenger->new();
$messenger->set_outgoing_window(10);
my $message = qpid::proton::Message->new();

my $address = $ARGV[0];
$address = "0.0.0.0" if !defined $address;
$message->set_address($address);
# how do we set properties and body?

$messenger->start();
my $tracker = $messenger->put($message);
print "Put: $message\n";
$messenger->send();
print "Status: ", $messenger->status($tracker), "\n";
$messenger->stop();

File: tests/smoke/send.py
#!/usr/bin/env python
from __future__ import print_function
import sys
from proton import *

messenger = Messenger()
messenger.outgoing_window = 10
message = Message()

address = "0.0.0.0"
if len(sys.argv) > 1:
  address = sys.argv[1]

message.address = address
message.properties = {u"binding": u"python",
                      u"version": sys.version}
message.body = u"Hello World!"

messenger.start()
tracker = messenger.put(message)
print("Put: %s" % message)
messenger.send()
print("Status: %s" % messenger.status(tracker))
messenger.stop()
